host certs: pass the remote address along in the request

zmb3 opened the pull request:
This mirrors the behavior that already exists in the HTTP API.

zmb3 committed:
host certs: pass the remote address along in the request

This mirrors the behavior that already exists in the HTTP API.

zmb3 commented:
Preserving the behavior that we already have in the api server: https://github.com/gravitational/teleport/blob/master/lib/auth/auth.go#L1764

Open to suggestions on how to test this. Our existing tests for the GRPC server don't cover this use case, and I wanted to get the fix out.

r0mant reviewed (commented):
> Open to suggestions on how to test this. Our existing tests for the GRPC server don't cover this use case, and I wanted to get the fix out.

@zmb3 I would exercise the GenerateHostCerts call (as one of grpcserver_test.go tests) and passed it `0.0.0.0` among AdditionalPrincipals to trigger the code path that replaces it from RemoteAddr. Then the test should fail without your fix and pass with it.

zmb3 committed:
Add test for GenerateHostCerts

We include 0.0.0.0 in additional principals in order to verify
that the logic for replacing that with RemoteAddr is working
correctly.

zmb3 committed:
Merge branch 'master' into zmb3/populate-remote-addr

codingllama reviewed (approved):
LGTM.
